src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -2671,20 +3159,43 @@ void M3()
         S_WithSemiAutoProperty s2 = s1;
     }
 }
-");
+";
+            var comp = CreateCompilation(new[] { source1, source2 });
             var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
             comp.TestOnlyCompilationData = accessorBindingData;
             Assert.Equal("System.Int32 S_WithAutoProperty.<P>k__BackingField", comp.GetTypeByMetadataName("S_WithAutoProperty").GetMembers().OfType<FieldSymbol>().Single().ToTestDisplayString());
             Assert.Empty(comp.GetTypeByMetadataName("S_WithManualProperty").GetMembers().OfType<FieldSymbol>());
             Assert.Empty(comp.GetTypeByMetadataName("S_WithSemiAutoProperty").GetMembers().OfType<FieldSymbol>());
+
+            if (structTreeFirst)
+            {
+                comp.GetDiagnosticsForSyntaxTree(CompilationStage.Compile, comp.SyntaxTrees[0], filterSpanWithinTree: null, includeEarlierStages: true).Verify();
+            }
+
+            comp.GetDiagnosticsForSyntaxTree(CompilationStage.Compile, comp.SyntaxTrees[1], filterSpanWithinTree: null, includeEarlierStages: true).Verify(
+                // (7,33): error CS0165: Use of unassigned local variable 's1'
+                //         S_WithAutoProperty s2 = s1;
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "s1").WithArguments("s1").WithLocation(7, 33),
+                // (19,37): error CS0165: Use of unassigned local variable 's1'
```

Comment:
>// (19,37): error CS0165: Use of unassigned local variable 's1' [](http://example.com/codeflow?start=16&length=63)

Is there a way to make this error to go away without using the struct constructor?

Reply:
I think `s1 = default;` should let this error go. I'll confirm with a test.

Is that what you meant?

Reply:
This is one way, I guess. Any other ways. Let's add test for all options

Reply:
@AlekseyTs I couldn't think of ways other than using the struct constructor or assigning to default. The only other way I could think of is assigning to `field` directly, but that's not an option.